linopy/model.py

```diff
@@ -1134,20 +1139,25 @@ def solve(
                 f"Keyword argument `io_api` has to be one of {IO_APIS} or None"
             )
 
-        if remote:
-            solved = remote.solve_on_remote(
-                self,
-                solver_name=solver_name,
-                io_api=io_api,
-                problem_fn=problem_fn,
-                solution_fn=solution_fn,
-                log_fn=log_fn,
-                basis_fn=basis_fn,
-                warmstart_fn=warmstart_fn,
-                keep_files=keep_files,
-                sanitize_zeros=sanitize_zeros,
-                **solver_options,
-            )
+        if remote or oetc_settings:
+            if remote and oetc_settings:
+                raise ValueError("Remote and OETC can't be active at the same time")
+            if remote:
+                solved = remote.solve_on_remote(
+                    self,
+                    solver_name=solver_name,
+                    io_api=io_api,
+                    problem_fn=problem_fn,
+                    solution_fn=solution_fn,
+                    log_fn=log_fn,
+                    basis_fn=basis_fn,
+                    warmstart_fn=warmstart_fn,
+                    keep_files=keep_files,
+                    sanitize_zeros=sanitize_zeros,
+                    **solver_options,
+                )
+            else:
+                solved = OetcHandler(oetc_settings).solve_on_oetc(self)  # type: ignore
```

Comment:
@KristijanFaust-OET I rather meant here, if we could do something like:


```suggestion
        if remote is not None:
            if isinstance(remote, OetHandler):
                solved = OetcHandler(oetc_settings).solve_on_oetc(self)  # type: ignore
            else:
                solved = remote.solve_on_remote(
                    self,
                    solver_name=solver_name,
                    io_api=io_api,
                    problem_fn=problem_fn,
                    solution_fn=solution_fn,
                    log_fn=log_fn,
                    basis_fn=basis_fn,
                    warmstart_fn=warmstart_fn,
                    keep_files=keep_files,
                    sanitize_zeros=sanitize_zeros,
                    **solver_options,
                )
```

that would enrich the `remote` argument. does that make sense?

Reply:
Yes exactly, but that also requires all the changes I mentioned in order to work. 

Maybe I'm missing something but I don't see a pre-step in linopy where the remote argument is constructed from the config settings before being passed to the solve function. So I guess it is expected from the client to construct that, right?

Reply:
ah yes! now I understand where you are coming from! how about we keep the oetc settings in the config in pypsa-eur and if non-empty a OetcHandler is created before the solve function, so in pypsa-eur this would mean


```
   oetc = solving.get("oetc", None)
    if oetc:
        oetc["credentials"] = OetcCredentials(
            email=os.environ["OETC_EMAIL"], password=os.environ["OETC_PASSWORD"]
        )
        oetc["solver"] = kwargs["solver_name"]
        oetc["solver_options"] = kwargs["solver_options"]
        oetc_settings = OetcSettings(**oetc)
        oetc_handler = OetcHandler(oetc_settings)
        kwargs["remote"] = oetc_handler
```
Does that make sense? 

I would like to keep the argument in the solve function abstract from oetc. I am also happy to introduce another argument like `service`/`cloud_solver` or so. But note that from a user perspective remote and oetc do a similar thing, they take the model and send it to a server to solve it there, even though the underlying architecture is very different.

Reply:
These changes are exactly what I'm talking about. But if we do this now on this branch we will break compatibility with Agora's implementation which is already shipped to them.

@siddharth-krishna 
Is it okay if we do the update now, notify Agora and send them an update?

Reply:
I see, if it is already in use, I don't want to be an extra hurdle (given the ages it took until the review). we could also come up with a follow up and a proper deprecation if you want

Reply:
This change is reasy to do, and I think it is not worth to have multiple branches/PRs for it, if we wouldn't have Agora in the mix. Lets wait to see what @siddharth-krishna, if he is ok with it, I'll do them in this PR, and we'll update Agora's codebase too.